jetbrains-core/src/software/aws/toolkits/jetbrains/services/schemas/code/SchemaCodeDownloader.kt

```diff
@@ -66,20 +68,27 @@ class SchemaCodeDownloader(
     }
 
     companion object {
-        fun create(project: Project): SchemaCodeDownloader = SchemaCodeDownloader(
-            CodeGenerator(project.awsClient()),
-            CodeGenerationStatusPoller(project.awsClient()),
-            CodeDownloader(project.awsClient()),
-            CodeExtractor(),
-            ProgressUpdater()
-        )
+        fun create(project: Project): SchemaCodeDownloader {
+            val connectionSettings = AwsConnectionManager.getInstance(project).connectionSettings()
+                ?: throw IllegalStateException("Attempting to use SchemaCodeDownload without valid AWS connection")
+            return create(connectionSettings)
+        }
+
+        fun create(connectionSettings: ConnectionSettings): SchemaCodeDownloader {
+            val clientManager = AwsClientManager.getInstance()
+            return SchemaCodeDownloader(
+                CodeGenerator(clientManager.getClient(connectionSettings.credentials, connectionSettings.region)),
+                CodeGenerationStatusPoller(clientManager.getClient(connectionSettings.credentials, connectionSettings.region)),
+                CodeDownloader(clientManager.getClient(connectionSettings.credentials, connectionSettings.region)),
```

Comment:
does `getClient` not have an override with  `connectionSettings` function? if not, we should add it

Reply:
Nope, planned on it with Kyle, (it is why ResourceSelector has one), but I didn't add it to client manager yet 

(we will deprecate the non connection settings ones)

Reply:
Remembered why I have not done this yet, ConnecionSettings needs to be moved from jetbrains-core to core first.